File: app/src/main/java/com/dublikunt/nclient/settings/Favorites.java
package com.dublikunt.nclient.settings;

import androidx.annotation.NonNull;

import com.dublikunt.nclient.api.components.Gallery;
import com.dublikunt.nclient.api.components.GenericGallery;
import com.dublikunt.nclient.async.database.Queries;

public class Favorites {
    public static boolean addFavorite(Gallery gallery) {
        Queries.FavoriteTable.addFavorite(gallery);
        return true;
    }

    public static boolean removeFavorite(@NonNull GenericGallery gallery) {
        Queries.FavoriteTable.removeFavorite(gallery.getId());
        return true;
    }

    public static boolean isFavorite(GenericGallery gallery) {
        if (gallery == null || !gallery.isValid()) return false;
        return Queries.FavoriteTable.isFavorite(gallery.getId());
    }
}

File: app/src/main/kotlin/com/dublikunt/nclient/settings/Favorites.kt
package com.dublikunt.nclient.settings

import com.dublikunt.nclient.api.gallerys.Gallery
import com.dublikunt.nclient.api.gallerys.GenericGallery
import com.dublikunt.nclient.async.database.Queries

object Favorites {
    fun addFavorite(gallery: Gallery): Boolean {
        Queries.FavoriteTable.addFavorite(gallery)
        return true
    }

    fun removeFavorite(gallery: GenericGallery): Boolean {
        Queries.FavoriteTable.removeFavorite(gallery.id)
        return true
    }

    fun isFavorite(gallery: GenericGallery?): Boolean {
        return if (gallery == null || !gallery.valid) false else Queries.FavoriteTable.isFavorite(
            gallery.id
        )
    }
}
